adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -283,79 +283,32 @@ private void handleSessionOpen(final ProtonConnection conn, final ProtonSession
     protected void handleRemoteReceiverOpen(final ProtonReceiver receiver, final ProtonConnection conn) {
         final Device authenticatedDevice = conn.attachments().get(AmqpAdapterConstants.KEY_CLIENT_DEVICE,
                 Device.class);
-        if (authenticatedDevice == null && receiver.getRemoteTarget() == null) {
-            // anonymous relay only supported for authenticated clients (gateways in particular)
-            receiver.setCondition(
-                    ProtonHelper.condition(AmqpError.NOT_ALLOWED, "anonymous relay ONLY supported for authenticated clients"));
+        validateEndpoint(receiver, authenticatedDevice).recover(t -> {
+            LOG.debug(
+                    "fail to establish a receiving link with client [{}] due to an invalid endpoint [{}]."
+                            + " closing the link",
+                    receiver.getName(), receiver.getRemoteQoS());
+            receiver.setCondition(AmqpContext.getErrorCondition((ServiceInvocationException) t));
```

Comment:
how do you know that it is a `ServiceInvocationException`? 
`AmqpContext.getErrorCondition()` should be able to do this check ...